### ISSUE TITLE
💅 Templated merged interface incorrectly identified as unused

### ISSUE THREAD

**Issue by sparr** [biomejs/biome#8331]
### Environment information

```bash
CLI:
  Version:                      2.3.8
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
(cut for brevity, 202 rules)

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedVariables

### Playground link

https://biomejs.dev/playground/?lintRules=noUnusedVariables&code=aQBuAHQAZQByAGYAYQBjAGUAIABBAHIAcgBhAHkAPABUAD4AIAB7AAoAIAAgAG4AZQB3AE0AZQB0AGgAbwBkADoAIABGAHUAbgBjAHQAaQBvAG4AOwAKAH0ACgAKAEEAcgByAGEAeQAuAHAAcgBvAHQAbwB0AHkAcABlAC4AbgBlAHcATQBlAHQAaABvAGQAPQAoACkAPQA%2BAHsAfQA7AA%3D%3D&language=ts

### Expected result

No Biome linter warning about `interface Array` being unused because removing that interface definition results in the new method assignment producing this typescript error: `Property 'newMethod' does not exist on type 'any[]'.ts(2339)`

PS: The same rule also triggers on `<T>` as being unused, while I believe it is necessary to avoid `All declarations of 'Array' must have identical type parameters.ts(2428)`, but I won't know for sure that that case still triggers until the primary report here is resolved.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by curatedcode** [comment]
The rule is correct. When you write Array.prototype.newMethod you’re extending the global Array object not the interface you created.

If you want typescript to recognize the new method, you need to augment the global type:
```
declare global {
	interface Array<T> {
		newMethod: (item: T) => void;
	}
}

Array.prototype.newMethod = <T>(item: T) => {}
```

**Comment by sparr** [comment]
In a file with no top level imports or exports, interfaces are applied to the global scope by default, and an explicit `declare global` is an error: `Augmentations for the global scope can only be directly nested in external modules or ambient module declarations.ts(2669)`

**Comment by curatedcode** [comment]
Got it you're treating it as a script not a module. Thanks for pointing that out
